test/deploy/rackhd_source_install.py

```diff
@@ -281,14 +213,11 @@ def test05_install_rackhd_config_files(self):
         fit_common.remote_shell('mkdir -p ~/src/on-http/static/swagger-ui')
 
     def test06_startup(self):
-        print "Start services."
-        startup = open('startup.sh', 'w')
-        startup.write('cd ~/;nf start&\n')
-        startup.close()
-        fit_common.scp_file_to_ora('startup.sh')
+        print "**** Start services."
         self.assertEqual(fit_common.remote_shell("chmod 777 startup.sh;/etc/init.d/isc-dhcp-server restart")['exitcode'], 0, "dhcp startup failure.")
-        self.assertEqual(fit_common.remote_shell("nohup ./startup.sh")['exitcode'], 0, "RackHD startup failure.")
+        self.assertEqual(fit_common.remote_shell("cd ~/;pm2 start rackhd-pm2-config.yml > /dev/null 2>&1")['exitcode'], 0, "RackHD startup failure.")
         print "**** Check installation."
+        fit_common.time.sleep(10)
```

Comment:
Yep, it failed when the sleep was not prior to the startup loop. Looks like an anomaly of the PM2 startup routine.